packages/editor/src/store/selectors.js

```diff
@@ -601,6 +601,19 @@ export function getBlockName( state, clientId ) {
 	return block ? block.name : null;
 }
 
+/**
+ * Returns whether a block is valid or not.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block client ID.
+ *
+ * @return {boolean} Is Valid.
+ */
+export function isBlockValid( state, clientId ) {
+	const block = state.editor.present.blocks.byClientId[ clientId ];
+	return block && block.isValid;
```

Comment:
If the block is undefined we will return undefined instead of a boolean. I think we should do `!! ( block && block.isValid )` or `!! block &&  block.isValid` if we are totally sure isValid is always defined to a boolean.